board/app/controllers/comment_controller.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+class CommentController extends AppController {
+
+const WRITE_COMMENT = 'write';
+const WRITE_COMMENT_END = 'write_end';
+const MAX_USER_IN_NEWSFEED = 1;
+
+    public function write() 
+    {
+        $thread_id = Param::get('thread_id');
+        $thread = Thread::get($thread_id);
+        $comment = new Comment();
+        $current_page = Param::get('page_next', self::WRITE_COMMENT);
+
+        switch($current_page) { 
+            case self::WRITE_COMMENT:
+                break;
+        
+            case self::WRITE_COMMENT_END:                
+                $params = array(
+                'body' => Param::get('body'),
+                'username' => $_SESSION['username'],
+                'user_id' => $_SESSION['user_id']
+                );
+                $comment = new Comment($params);
+                
+                try {            
+                    $comment->write($comment, $thread_id);
+                } catch (ValidationException $e) {                    
+                    $current_page = self::WRITE_COMMENT;
+                }    
+                break;
+            default:
+                throw new NotFoundException("{$current_page} is not found");
+                break;
+        }
+        $this->set(get_defined_vars());
+        $this->render($current_page);   
+    }
+
+    public function delete() 
+    { 
+        $comment = Comment::get(Param::get('comment_id'));
+        $comment->delete($_SESSION['user_id']);
+        $this->set(get_defined_vars());
+        $this->render('comment/delete');
+    }  
+
+    public function favorites() 
+    { 
+        $per_page = MAX_DATA_PER_PAGE;
+        $current_page = Param::get('page', 1);
+        $pagination = new SimplePagination($current_page, $per_page);
+
+        $id = $_SESSION['user_id'];
```

Comment:
Perhaps for consistency it's better to name this variable like the others as
```
$user_id
```
:bow: